.github/workflows/check_pr_title.yml

```diff
@@ -0,0 +1,49 @@
+name: Check PR title
+
+on:
+  pull_request:
+    # By default, a workflow only runs when a pull_request's activity type is opened, synchronize, or reopened. We
+    # explicitly override here so that PR titles are re-linted when the PR text content is edited.
+    #
+    # Possible values: https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request
+    types: [opened, edited, reopened, synchronize, ready_for_review]
+
+jobs:
+  pr-lint:
+    permissions:
+      pull-requests: write
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: morrisoncole/pr-lint-action@51f3cfabaf5d46f94e54524214e45685f0401b2a
+        if: github.actor != 'dependabot[bot]'
+        with:
+          title-regex: "^(Revert \")?(\\[\\S.*\\]\\s?)+\\s+[a-zA-Z`].*"
+          repo-token: "${{ secrets.GITHUB_TOKEN }}"
+          on-failed-regex-fail-action: true
+          on-failed-regex-create-review: true
+          on-failed-regex-request-changes: true
+          on-failed-regex-comment:
+            "The PR title does not conform to the '[\\<Project\\>] Title' format.
+            Please update the PR title.
+            \n\n
+            Typical [\\<Project\\>] values include:
+            \n\n
+            * [stdlib] — indicates a change to the Mojo standard library code\n
+            * [docs] — indicates a change to the documentation
```

Comment:
I think using code font here will help emphasize that the square brackets are actually important:
```suggestion
            * `[stdlib]` — indicates a change to the Mojo standard library code\n
            * `[docs]` — indicates a change to the documentation
```